src/RA_Interface.cpp

```diff
@@ -523,13 +556,13 @@ void RA_Init(HWND hMainHWND, int nConsoleID, const char* sClientVersion)
 {
     const char* sVerInstalled = _RA_InstallIntegration();
 
-    const char* sHostName = nullptr;
+    char sHostName[256] = "";
     if (_RA_HostName != nullptr)
-        sHostName = _RA_HostName();
```

Comment:
This is what was causing the 12005 error. `_RA_HostName()` was returning a pointer to data in the DLL, so when the DLL gets unloaded to be replaced by the new one, the `sHostName` pointer was no longer pointing at valid data. If the valid data is replaced between lines 575 and 588, the host name may be garbage, which results in the 12005 error.